src/module.c

```diff
@@ -4922,21 +4956,7 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
     replicate = flags & REDISMODULE_ARGV_REPLICATE;
     va_end(ap);
 
-    /* Setup our fake client for command execution. */
-    if (server.module_client == NULL) {
-        /* This is the first RM_Call() ever. Create reusable client. */
-        c = server.module_client = createClient(NULL);
-    } else if (server.module_client->argv == NULL) {
-        /* The reusable module client is not busy with a command. Use it. */
-        c = server.module_client;
-    } else {
-        /* The reusable module client is busy. (It is probably used in a
-         * recursive call to this module.) */
-        c = createClient(NULL);
-    }
-
-    c->user = NULL; /* Root user. */
-    c->flags = CLIENT_MODULE;
+    c = moduleAllocTempClient();
```

Comment:
Why not continue using `server.module_client` here instead of using the cache (which require mutex locking)?

Reply:
It's just to handle temporary clients from a single place and clear code a bit. `moduleFreeContextReusedClient` change is same as well. Yes, there might be performance hit with these changes. Even without using cache here, clearClient() is doing more compared the previous partial clear operation. I'm not sure how sensitive modules are to these changes. 

Just to avoid the risk, I can revert these two back to using the same client instance at least. What do you think?